internal/provider/resource_application.go

```diff
@@ -222,10 +222,6 @@ func resourceApplicationRead(ctx context.Context, d *schema.ResourceData, meta i
 		return nil
 	}
 
-	// TODO: This is a temporary fix to preserve the defined charm channel, as we cannot currently pull this from the API
-	// The if exists statement is also required to allow import to function when no exiting data is in the state
-	// Remove these lines and uncomment under the next TODO
-
 	var charmList map[string]interface{}
 	_, exists := d.GetOk("charm")
 	if exists {
```

Comment:
Do we not also need to specify `charmList["channel"] = response.Channel` here? If I'm understanding it correctly this would happen if someone changes a channel outside of terraform but the application is already in terraform state and they're doing an import.

Reply:
You're right.